dcm4che-core/src/main/java/org/dcm4che3/io/DicomOutputStream.java

```diff
@@ -170,6 +170,9 @@ public void writeHeader(int tag, VR vr, int len) throws IOException {
         ByteUtils.tagToBytes(tag, b, 0, bigEndian);
         int headerLen;
         if (!TagUtils.isItem(tag) && explicitVR) {
+            if ((len & 0xffff0000) != 0 && vr.headerLength() == 8) {
+                vr = VR.UN;
+            }
```

Comment:
Add a comment in to clarify what you're doing here:  DCM4CHE has a lot of bit masks and binary math :)

Reply:
Ok Andrew i've cherry picked Gunther's change from master and added a comment to it.